lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -141,6 +141,143 @@ LogicalResult ReturnOp::verify() {
 
   return success();
 }
+
+//===----------------------------------------------------------------------===//
+// InstanceOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult InstanceOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto targetClass = getClass(&symbolTable);
+  if (!targetClass)
+    return emitOpError() << "'" << getClassName() << "' does not exist";
+
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// PortReadOp
+//===----------------------------------------------------------------------===//
+
+template <typename TSrcOp, typename TTargetOp>
+LogicalResult verifyPortSymbolUses(
+    TSrcOp op, StringAttr symName, llvm::function_ref<Type(TSrcOp)> getPortType,
+    SymbolTableCollection &symbolTable,
+    llvm::function_ref<FailureOr<TTargetOp>(Operation *)> getTargetOp) {
+
+  ClassOp parentClass = op->template getParentOfType<ClassOp>();
+  if (!parentClass)
+    return op->emitOpError()
```

Comment:
Should this be an assertion? The parent op should be checked by the ops verifier, correct?